crates/celo-revm/src/handler.rs

```diff
@@ -131,16 +173,9 @@ where
         let caller = ctx.tx().caller();
 
         // Convert costs to fee currency
-        let fee_currency_context = &evm.ctx().chain().fee_currency_context;
-        let base_fee_in_erc20 = fee_currency_context
-            .celo_to_currency(fee_currency, U256::from(basefee))
-            .map_err(|e| ERROR::from_string(e))?;
-        // Convert base_fee_in_erc20 (U256) to u128 for gas price calculations
-        let base_fee_in_erc20_u128: u128 = base_fee_in_erc20
-            .try_into()
-            .expect("Failed to convert base_fee_in_erc20 to u128: value exceeds u128 range");
-        let effective_gas_price = evm.ctx().tx().effective_gas_price(base_fee_in_erc20_u128);
-        let tip_gas_price = effective_gas_price.saturating_sub(base_fee_in_erc20_u128);
+        let base_fee_in_erc20 = self.cip64_get_base_fee(evm, fee_currency, basefee)?;
+        let effective_gas_price = evm.ctx().tx().effective_gas_price(base_fee_in_erc20);
+        let tip_gas_price = effective_gas_price.saturating_sub(base_fee_in_erc20);
```

Comment:
If we want to be really safe, we can use a checked_sub here, so that we'll notice if we introduce a similar problem again.